content/tutorials/manage-the-rippled-server/configuration/configure-full-history.md

```diff
@@ -0,0 +1,98 @@
+
+You do not need a full history server to participate in the network, validate transactions, or know the current state of the network. Full history is only useful for knowing the outcome of transactions that occurred in the past, or the state of the ledger at a given time in the past. To get such information, you must rely on other servers having the history you need.
+
+If you want to contribute to storing the history of the XRP Ledger network without storing the full history, you can [configure history sharding](configure-history-sharding.html) to store randomly-selected chunks of ledger history instead.
+
+## Configuration Steps
+
+To configure your server to acquire and store full history, complete the following steps:
+
+1. Stop the `rippled` server if it is running.
+
+        $ sudo systemctl stop rippled
+
+0. Remove (or comment out) the `online_delete` and `advisory_delete` settings from the `[node_db]` stanza of your server's config file:
+
+        [node_db]
+      	type=RocksDB
+      	path=/var/lib/rippled/db/rocksdb
+      	open_files=2000
+      	filter_bits=12
+      	cache_mb=256
+      	file_size_mb=8
+      	file_size_mult=2
+        compression=1
+      	#online_delete=2000
+      	#advisory_delete=0
+
+    You can store full history with either RocksDB or NuDB as the key-value store behind the ledger store. NuDB requires less RAM but RocksDB requires less disk space. For more information, see [Capacity Planning](capacity-planning.html).
+
+    If you are using RocksDB, add `compression=1` to enable Snappy compression, which reduces the disk space necessary to store full history (at a cost of greater CPU usage when reading or writing history).
+
+    {% include '_snippets/conf-file-location.md' %}<!--_ -->
+
+0. Set the `[ledger_history]` stanza of your server's config file to `full`:
+
+        [ledger_history]
+        full
+
+0. Set the `[ips_fixed]` stanza of your server's config file to explicitly peer with at least one server that has full history available.
+
+        [ips_fixed]
+        169.55.164.20
+        50.22.123.215
+
+    Your server can only download historical data from the peer-to-peer network if one its direct peers has the data available. The easiest way to ensure you can download full history is to peer with a server that already has full history.
+
+    **Tip:** Ripple makes a pool of full history servers publicly available. You can resolve the domain `s2.ripple.com` a few times to get the IP addresses of these servers. Ripple offers these servers as a public service, so be aware that their availability to peer with other servers is limited and you may be blocked if you abuse them.
+
+0. If you have a database dump from another full-history server to use as a basis, set the `[import_db]` stanza of your server's config file to point to the data to be imported. (Otherwise, skip this step.)
+
+        [import_db]
+      	type=RocksDB
+      	path=/tmp/full_history_dump/
```

Comment:
The syntax for the `[import_db]` stanza was not clearly specified in the config file so I took an educated guess. Please confirm.

Reply:
I know this is just an example and not sure if this matters - but maybe the `type` should be set to `NuDB`, just to reinforce our recommendation that you use NuDB for full history?

Reply:
Thank you for writing this tutorial, Rome - great info, so concise, and I think the topic is of great interest.

Reply:
Good catch on the RocksDB thing. I changed the `[node_db]` stanza but missed the later `[import_db]` example on my first pass. I think technically they _can_ be different types, but most likely if you're importing from another DB that has full history, it'll be NuDB→NuDB.